pgml-dashboard/notebooks/templates/notebooks/cell.html

```diff
@@ -1,61 +1,82 @@
+  <div data-controller="cell" class="notebook-cell {{ cell.get_cell_type_display|lower }}">
+
+    {% if edit %}
+    <!-- Edit -->
+    <div class="notebook-cell-edit">
+      <form action="{% url 'notebooks/cell/edit' notebook.pk cell.pk %}" method="post" data-cell-target="form">
+        <div class="flex">
+          {% csrf_token %}
+          <div class="notebook-cell-code flex-grow">
+            <textarea name="contents" data-cell-target="editor">{{ cell.contents }}</textarea>
+          </div>
+          <div class="flex flex-end">
+            <select name="cell_type" data-action="cell#selectCellType" title="Select cell type">
+              <option value="3" {% if cell.cell_type == 3 %}selected{% endif %}>SQL</option>
+              <option value="1" {% if cell.cell_type == 1 %}selected{% endif %}>Markdown</option>
+            </select>
+            <button type="submit" class="button notebook-button" title="Execute cell">
+              <span class="material-symbols-outlined">play_arrow</span>
+            </button>
+          </div>
         </div>
-      </div>
-    </form>
-  </div>
+      </form>
+    </div>
+    {% endif %}
+
+    <!-- Display -->
+    <div>
+      <div class="flex">
+        <div class="cell-number">{{ cell.cell_number }}</div>
+        <div class="flex-grow">
+          {% if cell.code %}
+          <div class="notebook-contents">
+            <pre><code class="language-sql">{{ cell.contents }}</code></pre>
+          </div>
+          {% endif %}
 
-  <!-- Display -->
-  <div>
-    <div class="flex">
-      <div class="cell-number">{{ cell.cell_number }}</div>
-      <div class="flex-grow">
+          {% if cell.html %}
+            <div class="notebook-rendering">
+              {{ cell.html }}
+            </div>
+          {% endif %}
+        </div>
+          
         {% if cell.code %}
-        <div class="notebook-contents">
-          <pre><code class="language-sql">{{ cell.contents }}</code></pre>
+        <div class="notebook-button margin-left-1">
+          <form action="{% url 'notebooks/cell/play' notebook.pk cell.pk %}" method="post">
+            {% csrf_token %}
+            <button type="submit" title="Execute cell">
+              <span class="material-symbols-outlined">
+                play_arrow
+              </span>
+            </button>
+          </form>
         </div>
         {% endif %}
-
-        <div class="notebook-rendering">
-          {{ cell.html }}
+        <div class="notebook-button">
+          <form action="{% url 'notebooks/cell/remove' notebook.pk cell.pk %}" method="post">
+            {% csrf_token %}
+            <button type="submit" title="Delete cell">
+              <span class="material-symbols-outlined">delete_forever</span>
+            </button>
+          </form>
+        </div>
+        <div class="notebook-button">
+          <form action="{% url 'notebooks/cell/edit' notebook.pk cell.pk %}?bust_cache={{ bust_cache }}" method="get">
```

Comment:
If you have any ideas how to get around this using a cleaner method, that would be great. The problem here is when we submit the `GET` form, Turbo annotates the frame as coming from `notebooks/<pk>/cell/<pk>/edit` URL  which won't change. When we edit the cell again, Turbo doesn't execute the fetch because it thinks the result will be the same, so editing doesn't work twice on the same cell. Adding a unique parameter to the URL "busts that cache": Turbo can't make any assumptions about the content and executes the fetch.